dvc/stage.py

```diff
@@ -313,7 +315,7 @@ def _stage_fname(cls, fname, outs, add):
 
         fname = path.basename(out.path) + cls.STAGE_FILE_SUFFIX
 
-        if add and out.is_local:
+        if add and out.is_local and not contains_link(out.path):
```

Comment:
Imagine this scenario. Your repo is located under symlink_dir. E.g. `/path/to/symlink_dir/repo`. Now you do `dvc add data_dir/foo`(data_dir is not a symlink), which results in `contains_links('/path/to/symlink_dir/repo/data_dir/foo')`, which will return `True` and will make `foo.dvc` appear in the current directory instead of behaving as it should. I'm sure we will run into this when running tests on mac, since its `/tmp` is a symlink. How should we go about solving this? Ideas? :slightly_smiling_face:  Also, let's add a test for such a scenario.

Reply:
Good point!

Reply:
As discussed offline: We will check whether symlink is below repository root directory.